player/command.c

```diff
@@ -2773,6 +2773,21 @@ static void update_hidpi_window_scale(struct MPContext *mpctx, bool hidpi_scale)
     vo_control(vo, VOCTRL_SET_UNFS_WINDOW_SIZE, s);
 }
 
+static int mp_property_ambient_light_lux(void *ctx, struct m_property *prop,
+                                         int action, void *arg)
+{
+    MPContext *mpctx = ctx;
+    struct vo *vo = mpctx->video_out;
+    if (!vo)
+        return M_PROPERTY_UNAVAILABLE;
+
+    int lux;
+    if (vo_control(vo, VOCTRL_GET_AMBIENT_LUX, &lux) < 1)
+        return M_PROPERTY_UNAVAILABLE;
+
+    return m_property_int_ro(action, arg, lux);
```

Comment:
Why is this `int`? Wouldn't make sense to make it `float`?

Reply:
it's int because [VOCTRL_GET_AMBIENT_LUX](https://github.com/mpv-player/mpv/blob/8d20b726537fb98cf509bb156f68c9a3769d2001/video/out/vo.h#L118) is already int (since reusing the same VOCTRL as the old mechanic) and i didn't want to change that. though if you prefer i can change it and all the dependent code.
i am not sure if float makes sense (contextual)/the precision is needed?

Reply:
I don't know what vlaues typical apple display can report, but lux itself, has some range <1, where we wouldn't have resolution for low lighting to show. 
![image](https://github.com/user-attachments/assets/528da066-d4af-4bd6-a6e6-d823be0f8c6f)

Also Firefox code that is referenced in conversion, is using float, see https://github.com/mozilla/gecko-dev/blob/master/hal/cocoa/CocoaSensor.mm#L22-L38

I wanted to discuss this, because now that it is exported in public api, we may have hard time changing it in the future if needed.

Reply:
yeah, i will change it to float/double. in the end it doesn't hurt to have a higher precision.